lib/OpenQA/WebAPI/Controller/Main.pm

```diff
@@ -30,6 +30,12 @@ sub _group_result {
 
     my %res;
     my $jobs = $group->jobs->search({"me.t_created" => $timecond,});
+    # arbitrary default limit
+    my $limit_jobs = $self->param('limit_jobs') // 4000;
```

Comment:
This still doesn't limit the jobs if `limit_jobs` is provided to be too big. And why checking the limit after the DB query? You can add something like `rows => $limit_jobs` to search call.